web_services/foxx/brightid/db.js

```diff
@@ -767,6 +788,70 @@ function getTestblocks(app, contextId) {
   }).toArray().map(b => b.action);
 }
 
+function getContextIds(coll) {
+  return coll.all().toArray().map(c => {
+    return {
+      user: c.user,
+      contextId: c.contextId,
+      timestamp: c.timestamp
+    }
+  });
+}
+
+function loadGroup(groupId) {
+  return query`RETURN DOCUMENT(${groupsColl}, ${groupId})`.toArray()[0];
+}
+
+function groupInvites(groupId) {
+  return invitationsColl.byExample({
+    "_to": 'groups/' + groupId,
+  }).toArray().filter(invite => {
+    return Date.now() - invite.timestamp < 86400000
+  }).map(invite => {
+    return {
+      inviter: invite.inviter,
+      invitee: invite._from.replace('users/', ''),
+      id: invite._key,
+      data: invite.data,
+      timestamp: invite.timestamp
+    }
+  });
+}
+
+function removePasscode(contextKey) {
+  contextsColl.update(contextKey, {
+    passcode: null
+  });
+}
+
+function updateGroup(admin, groupId, url, timestamp) {
+  if (! groupsColl.exists(groupId)) {
+    throw 'group not found';
+  }
+  const group = groupsColl.document(groupId);
+  if (! group.admins || ! group.admins.includes(admin)) {
+    throw 'only admins can update the group';
+  }
+  groupsColl.update(group, {
+    url,
+    timestamp
+  });
+}
+
+function addSubkey(id, subkey, timestamp) {
+  const subkeys = usersColl.document(id).subkeys || [];
+  if (subkeys.indexOf(subkey) == -1) {
+    subkeys.push(subkey);
+    usersColl.update(id, { subkeys });
+  }
+}
+
+function removeSubkey(id, subkey) {
+  let subkeys = usersColl.document(id).subkeys || [];
+  subkeys = subkey ? subkeys.filter(s => s != subkey) : [];
+  usersColl.update(id, { subkeys });
+}
```

Comment:
It's clearer to have a second function called `removeAllSubkeys` than overload the `subKey` parameter with the meaning that if it's falsey, it should remove all the subKeys.

Reply:
Also see this change I made in the comment of the issue https://github.com/BrightID/BrightID-Node/issues/155 that says the all signing keys are on the same level.  We could use an array of `signingKeys` instead of a single `signingKey` in the users documents.

Reply:
The remove all subkeys (remove all signing keys) functionality is removed and we can handle it client-side by using the removeSigningKey function